finger-exercises-react/react/ecmascript/01 - spread/index.js

```diff
@@ -1,14 +1,44 @@
 import isArray from './utils';
 
-export function min(...args) {
-  if (isArray(args)) {
-    if (args.length > 1) {
-      return Math.min(args);
+function hasNestedArray(array) {
+  return isArray(array[0]);
+}
+
+function isEmpty(array) {
+  return array.length === 0;
+}
+
+export function min(...array) {
+  if (!isEmpty(array)) {
+    if (hasNestedArray(array)) {
+      return Math.min(...array[0]);
     }
+
+    return Math.min(...array);
   }
-  return args[0];
+
+  return undefined;
```

Comment:
I think if you don't return anything it already returns undefined. You don't need to return undefined explicitly

Reply:
@LucasZibell I see. I tried that but the linter throws the following error: **Expected to return a value at the end of function 'min'** 

I think that's to ensure that the function actually returns something. Do I delete that statement anyway ?

Reply:
Yeah eslint needs to have a return at the end of all the possible paths if there is a return in one of them. So as you say, it ensures that you are returning something in all the paths. However is nice for you to see and know that you don't have to return explicitly undefined and if you don't return anything, it already counts as undefined.

Reply:
Got it! Indeed it is. I didn't know that. I'll keep it in mind. Thanks Lucas!